tests/file_system_cache_test.py

```diff
@@ -56,5 +56,69 @@ def test_empty_key_get(self):
      with self.assertRaisesRegex(ValueError , r"key cannot be empty"):
        cache.get("")
 
+  def test_size_of_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir)
+      cache.put("foo", b"bar")
+      self.assertEqual(cache._get_cache_directory_size(), 3)
+
+  def test_size_of_empty_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir)
+      self.assertEqual(cache._get_cache_directory_size(), 0)
+
+  def test_size_of_existing_directory(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache1 = FileSystemCache(tmpdir)
+      cache1.put("foo", b"bar")
+      del cache1
+      cache2 = FileSystemCache(tmpdir)
+      self.assertEqual(cache2._get_cache_directory_size(), 3)
+
+  def test_cache_is_full(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=5)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      self.assertEqual(cache.get("foo"), None)
+
+  def test_file_removed(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=5)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      self.assertEqual(cache.get("second"), b"two")
+
+  def test_delete_multiple_files(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=6)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      cache.put("third", b"three")
+      self.assertEqual(cache.get("second"), None)
+
+  def test_big_file_exists(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=6)
+      cache.put("foo", b"bar")
+      cache.put("second", b"two")
+      cache.put("third", b"three")
+      self.assertEqual(cache.get("third"), b"three")
+
+  def test_file_bigger_than_cache(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir, max_cache_size_bytes=2)
+      with self.assertRaisesRegex(AssertionError, r"This compilation file is larger than the size of the cache"):
+        cache.put("foo", b"bar")
+
+  def test_directory_same_after_deletion_from_cache(self):
```

Comment:
I actually think you can delete this test case :D You already test in other places that the cache directory doesn't change across `FileSystemCache` instances, so I don't think you need to test every possible variation of this pattern. In general more test cases are better, but having too many redundant test cases can make it hard to read the test file later to understand what's being tested and what's not.